pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -118,9 +118,14 @@ func (p *staticPolicy) Name() string {
 }
 
 func (p *staticPolicy) Start(s state.State) error {
-	if err := p.validateState(s); err != nil {
-		klog.Errorf("[cpumanager] static policy invalid state: %v, please drain node and remove policy state file", err)
-		return err
+	err := p.validateState(s)
+	if err != nil {
+		klog.Errorf("[cpumanager] static policy invalid state: %v", err)
+		// if validating failed, clear state and set all cpus to shared pool
+		klog.Warningf("cpumanager cannot guarantee sane CPU affinity for existing containers unless drain this node and remove policy state file")
```

Comment:
I don't think this is reasonable as a default policy.  There may be high-performance applications which rely on exclusive access to CPUs, and they would rather fail entirely (and have the work taken over by other nodes) instead of run at less than full speed.

So if the state file is wrong, I think we need to fail and let something else clean it up and sort it out.

Reply:
Here it just clear the corrupted state (log a warning), and then reconciler will make consistence between running pods and cpuset cgroups (also write the correct state back to checkpoint) to ensure the high-performance apps be bound to assigned CPUs.